src/fosslight_scanner/fosslight_scanner.py

```diff
@@ -255,60 +264,113 @@ def init(output_path=""):
     return os.path.isdir(_output_dir), output_root_dir, result_log
 
 
+def check_compare_output_file(output_file_name, format):
+    CUSTOMIZED_FORMAT_FOR_COMPARE_MODE = {'excel': '.xlsx', 'html': '.html', 'json': '.json', 'yaml': '.yaml'}
+
+    if format == "":
+        logger.info(" * Default result format : excel")
+        format = "excel"
+
+    success, msg, output_path, output_file, output_extension = check_output_format(output_file_name, format,
+                                                                                   CUSTOMIZED_FORMAT_FOR_COMPARE_MODE)
+    success, final_excel_dir, result_log = init(output_path, False)
```

Comment:
In init(output_path, False) , the `success `value may be False because `make_outdir ` is False.

Isn't the `success `you want to check here the `success `value received when you call `check_output_format`?

Reply:
I changed it.